providers/view_provider.ts

```diff
@@ -0,0 +1,152 @@
+/*
+ * @adonisjs/view
+ *
+ * (c) AdonisJS
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+import { ApplicationService, HttpRouterService } from '@adonisjs/core/types'
+import { ViewContract } from '../types/index.js'
+import { BriskRoute, HttpContext } from '@adonisjs/core/http'
+
+/**
+ * View provider to register view to the application
+ */
+export default class ViewProvider {
+  constructor(protected app: ApplicationService) {}
+
+  /**
+   * Add globals for resolving routes
+   */
+  private addRouteGlobal(view: ViewContract, router: HttpRouterService) {
+    /**
+     * Adding `route` global
+     */
+    view.global('route', (routeIdentifier: string, params?: any, options?: any) => {
+      return router.makeUrl(routeIdentifier, params, options)
+    })
+
+    /**
+     * Adding `signedRoute` global
+     */
+    view.global('signedRoute', (routeIdentifier: string, params?: any, options?: any) => {
+      return router.makeSignedUrl(routeIdentifier, params, options)
+    })
+  }
+
+  /**
+   * Share application reference, a config and env variable with the
+   * templates.
+   */
+  private async addGlobals(view: ViewContract) {
+    const config = await this.app.container.make('config')
+
+    view.global('app', this.app)
+    view.global('config', (key: string, defaultValue?: any) => config.get(key, defaultValue))
+  }
+
+  /**
+   * Copy globals exposed by Edge
+   */
+  private async copyEdgeGlobals(view: ViewContract) {
+    const { GLOBALS } = (await import('edge.js')) as { GLOBALS: { [key: string]: any } }
+    Object.keys(GLOBALS).forEach((key) => view.global(key, GLOBALS[key]))
+  }
+
+  /**
+   * Registering the brisk route to render view directly from the route.
+   */
+  private registerBriskRoute() {
+    BriskRoute.macro('render', function renderView(this: BriskRoute, template: string, data?: any) {
+      return this.setHandler(({ view }: { view: ViewContract }) => {
+        return view.render(template, data)
+      })
+    })
+  }
+
+  /**
+   * Registering the http context getter to access an isolated
+   * view instance with the request and route.
+   */
+  private registerHTTPContextGetter(view: ViewContract) {
+    // Check if getter is already defined
+    if (Object.getOwnPropertyDescriptor(HttpContext.prototype, 'view')) {
```

Comment:
This check could be removed if we put `configurable: true` in the Object.defineProperty of `macroable` as explained on Discord